prov/verbs/src/ep_rdm/verbs_tagged_ep_rdm_states.c

```diff
@@ -547,6 +550,10 @@ fi_ibv_rdm_try_unexp_recv(struct fi_ibv_rdm_request *request,
 			if (request->parent) {
 				repost = fi_ibv_rdm_repost_multi_recv(request,
 					 found_request->len, rdata->ep);
+				if (!repost) {
+					ret = -FI_ENOMEM;
+					break;
+				}
 			}
```

Comment:
I don't think we want to abort here. We need to do something more with the unexpected message.